leshan-core/src/main/java/org/eclipse/leshan/core/util/TimestampUtil.java

```diff
@@ -0,0 +1,54 @@
+package org.eclipse.leshan.core.util;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.time.Instant;
+
+/**
+ * <p>
+ * This class contains methods for converting timestamps between <code>BigDecimal</code> and <code>Instant</code>
+ * format.
+ * </p>
+ */
+public class TimestampUtil {
```

Comment:
Maybe we can add some small unit test about it ? 

(especially to test the case of out of range bigdecimal ?)